### PR TITLE
[Postgres provider] Speed up feature insertion when pkid column is not set (fixes #18976)

### PR DESCRIPTION
**Pull request by rouault**
When inserting features whose attribute column corresponding to the pkid
is empty, and tat this pkid column has a default value using a sequence,
then we can remove it from the INSERT statement completely, which save us
from doing a SELECT nextval(...) for each row. On the provided test case
of that ticket, on a debug build, this cuts down the insertion time from 5 minutes
to 1 minute 10s.

### CODE REVIEW

File: src/providers/postgres/qgspostgresprovider.cpp [hunk 1]
```diff
@@ -2029,15 +2029,41 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
...
+            foundNonNullPK = true;
+            break;
+          }
+        }
+        skipSinglePKField = !foundNonNullPK;
+      }
+
+      if ( !skipSinglePKField )
+      {
+        Q_FOREACH ( int idx, mPrimaryKeyAttrs )
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresprovider.cpp]: range based for instead of Q_FOREACH

Comment (rouault) [reply src/providers/postgres/qgspostgresprovider.cpp]: ok, note that I didn't modify that code. Was just put inside a if ()

**Comment by nyalldawson** [comment]
Nice!

The biggest performance issue with postgres provider add features is that we don't features using copy. We should be able to detect if the FastInsert flag is set and in that case just throw all the features in via copy... That would speed up lots of postgres data operations, e.g. any method which exports/saves whole tables into postgres. But that's non trivial :)

**Comment by rouault** [comment]
> is that we don't features using copy.

To benchmark I tried with ogr2ogr as well (optimized build). In insert mode, it takes ~30 seconds. In copy mode: ~10 seconds

**Pull request merged**